gitdash/pages/api/single_repo.ts

```diff
@@ -0,0 +1,108 @@
+import { getSession } from "next-auth/client";
+import { Octokit } from "@octokit/core";
+import parse from "parse-link-header";
+
+export default async function GetDetails(
+  req: any,
+  res: {
+    status: (arg0: number) => {
+      (): any;
+      new (): any;
+      json: {
+        (arg0: {
+          user: any; 
+          repoName: any;  
+          repoData: any; 
+          contributors: any; 
+          languages: any; 
+          tags: any; 
+          branches:any; 
+          commits: any; 
+          forks: any; 
+        }): any;
+        new (): any;
+      };
+    };
+  }
+): Promise<any> {
+  const session = await getSession({ req });
+  const {repo} = req.query

```

Comment:
Think about setting up ESLint or something like that to encourage you to have a consistent code style.

Personally, I'm a fan of the [Airbnb style guide](https://github.com/airbnb/javascript).